### PR TITLE
Wishlist feature

### PR DESCRIPTION
**Pull request by Kevin-Solomon**
# WishList (ADD , DELETE ) Features
*To access these feature you have to be logged in*
```email : adarshbalika@gmail.com```
```password : adarshBalika123```

[CSB Link](https://g5ou9v.csb.app/)

### CODE REVIEW

File: src/context/wishlist/wishlistContext.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      } else {
+        const response = await axios.post(
+          '/api/user/wishlist',
+          { product },
+          { headers: { authorization: token } }
+        );
+        wishlistDispatch({
+          type: 'ADD_TO_CART',
+          payload: response.data.wishlist,
+        });
```
Comment (starver20) [inline src/context/wishlist/wishlistContext.js]: Here you should check if the response status is 200. Only then dispatch

File: src/context/wishlist/wishlistContext.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      }
+    });
+    return flag;
+  }
+  const deleteFromWishlist = async id => {
+    const response = await axios.delete(`/api/user/wishlist/${id}`, {
+      headers: { authorization: token.token },
+    });
+    wishlistDispatch({ type: 'DELETE', payload: response.data.wishlist });
+  };
```
Comment (starver20) [inline src/context/wishlist/wishlistContext.js]: Check for response here too.

**Review by AtulPant2704 — APPROVED** [review]
Just check for the response status after making network call, rest all looks good.

**Review by starver20 — APPROVED** [review]

File: src/context/wishlist/wishlistContext.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+          console.error(err);
+        }
+      }
+    }
+  };
+  function checkItem(state, itemToCheck) {
+    let flag = false;
+    state.forEach(item => {
+      if (item._id === itemToCheck._id) {
+        console.log('sike4');
```
Comment (mohammed-sas) [inline src/context/wishlist/wishlistContext.js]: you could remove the console log

**Review by mohammed-sas — APPROVED** [review]
well written code

File: src/context/wishlist/wishlistContext.js (hunk 3 above)
Comment (Kevin-Solomon) [reply src/context/wishlist/wishlistContext.js]: Thank you for pointing it out mohammed.

**Pull request merged**